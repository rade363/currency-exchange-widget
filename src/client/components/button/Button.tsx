import React from 'react';
import { OwnProps } from './types';
import './button.scss';

function Button({ children, active, styleType, className, ...otherProps }: OwnProps) {
  return (
    <button
      {...otherProps}
      type="button"
      className={`button button_${styleType} ${active ? 'button_active' : ''} ${className || ''}`}
      disabled={!active}
    >
      {children}
    </button>
  );
}

export default Button;
