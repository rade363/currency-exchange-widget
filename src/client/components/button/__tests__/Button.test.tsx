import React from 'react';
import { create } from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Button from '../Button';

describe('Button', () => {
  test('Should render active primary button', () => {
    const button = create(
      <Button
        styleType="primary"
        active
      >
        Click me
      </Button>,
    );
    const tree = button.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Active button should be active and clickable', () => {
    const mockCallBack = jest.fn();

    const { getByText } = render(
      <Button
        styleType="primary"
        active
        onClick={mockCallBack}
      >
        Click me
      </Button>,
    );

    fireEvent.click(screen.getByText(/click me/i));

    expect(getByText(/Click me/i).closest('button')).toBeEnabled();
    expect(mockCallBack).toHaveBeenCalledTimes(1);
  });

  test('Inactive button should be disabled and not fire the onClick event', () => {
    const mockCallBack = jest.fn();

    const { getByText } = render(
      <Button
        styleType="danger"
        active={false}
      >
        Click me
      </Button>,
    );

    fireEvent.click(screen.getByText(/click me/i));

    expect(getByText(/Click me/i).closest('button')).toBeDisabled();
    expect(mockCallBack).toHaveBeenCalledTimes(0);
  });
});
