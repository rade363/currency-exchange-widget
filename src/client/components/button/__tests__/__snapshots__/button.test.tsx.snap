// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render active primary button 1`] = `
<button
  className="button button_primary button_active "
  disabled={false}
  type="button"
>
  Click me
</button>
`;
